scripts/weather.py

```diff
@@ -0,0 +1,62 @@
+import requests, json
+from datetime import datetime, timedelta
+from deepmerge import always_merger
+from scripts.config import *
+from google.cloud import storage
+
+
+def get_weather_data(query, limit=30):
+    weather = {}
+    weather["forecast"] = {}
+    for day in range(limit, -1, -1):
+        date = datetime.today() - timedelta(days=day)
+        full_url = BASE_URL.format(API_KEY, query, date.strftime('%Y-%m-%d'))
+        response = requests.get(full_url)
+        data = response.json()
+        try:
+            forecast = data["forecast"]["forecastday"][0]
+            location = data["location"]
+            forecast["day"].pop("condition")
+            weather = {**weather, **location}
+
+            weather["forecast"][forecast["date_epoch"]] = forecast["day"]
+        except :
+            continue
+
+    return weather
+
+def __load_state_file(state):
```

Comment:
dunders are typically reserved for the Python Core API team. A single `_` prefix would suffice here. See https://amontalenti.com/2013/04/11/python-double-under-double-wonder